console/src/config.rs

```diff
@@ -0,0 +1,162 @@
+        group = "colors",
+        conflicts_with_all = &["no-colors", "truecolor"]
+    )]
+    palette: Option<Palette>,
+
+    #[clap(flatten)]
+    toggles: ColorToggles,
+}
+
+/// Toggles on and off color coding for individual UI elements.
+#[derive(Clap, Debug, Copy, Clone)]
+pub struct ColorToggles {
+    /// Disable color-coding for duration units.
+    #[clap(long = "no-duration-colors", parse(from_flag = std::ops::Not::not), group = "colors")]
+    pub(crate) color_durations: bool,
+
+    /// Disable color-coding for terminated tasks.
+    #[clap(long = "no-terminated-colors", parse(from_flag = std::ops::Not::not), group = "colors")]
+    pub(crate) color_terminated: bool,
+}
+
+// === impl Config ===
+
+impl Config {
+    pub fn trace_init(&mut self) -> color_eyre::Result<()> {
+        let filter = std::mem::take(&mut self.env_filter);
+        use tracing_subscriber::prelude::*;
+
+        // If we're on a Linux distro with journald, try logging to the system
+        // journal so we don't interfere with text output.
+        let journald = tracing_journald::layer().ok();
+
+        // Otherwise, log to stderr and rely on the user redirecting output.
+        let fmt = if journald.is_none() {
+            Some(
+                tracing_subscriber::fmt::layer()
+                    .with_writer(std::io::stderr)
+                    .with_ansi(atty::is(atty::Stream::Stderr)),
+            )
+        } else {
+            None
+        };
+
+        tracing_subscriber::registry()
+            .with(journald)
+            .with(fmt)
+            .with(filter)
+            .try_init()?;
+
+        Ok(())
+    }
+}
+
+// === impl Colors ===
+
+impl Colors {
+    /// Determines the color palette to use.
+    ///
+    /// The color palette is determined based on the following (in order):
+    /// - Any palette explicitly set via the command-line options
+    /// - The terminal's advertised support for true colors via the `COLORTERM`
+    ///   env var.
+    /// - Checking the `terminfo` database via `tput`
+    pub(crate) fn determine_palette(&self) -> Palette {
+        // Did the user explicitly disable colors?
+        if self.no_colors {
+            tracing::debug!("colors explicitly disabled by `--no-colors`");
+            return Palette::NoColors;
+        }
+
+        // Did the user explicitly select a palette?
+        if let Some(palette) = self.palette {
+            tracing::debug!(?palette, "colors selected via `--palette`");
+            return palette;
+        }
+
+        // Does the terminal advertise truecolor support via the COLORTERM env var?
+        if self.truecolor {
+            tracing::debug!("millions of colors enabled via `COLORTERM=truecolor`");
+            return Palette::All;
+        }
+
+        // Okay, try to use `tput` to ask the terminfo database how many colors
+        // are supported...
+        let tput = Command::new("tput").arg("colors").output();
```

Comment:
Separate point, it doesn't seem like `tputs` is available in Window CMD. I haven't checked if the console fully works in the console, but this might change that?

Reply:
Hmm, if the `tput` command fails, we'll just assume colors aren't supported. But, we could put the attempt to call out to `tput` behind `#[cfg(unix)]`, and not try to do it at all, if this won't work on Windows.

Do you know if there's a way to see if the console supports ANSI color escape codes on Windows? *Does* the windows console even support these escape codes?

Reply:
Oh I didn't notice that it handles `tput` failing. I guess that's fine. I know Windows terminals do support colors, though I believe through a different means. I remember `ripgrep` came up with a crate to make colors work on well on both... 🤷

Reply:
Okay, we should figure that out in a follow-up, I guess.